### PR TITLE
add record of unknown type test

### PR DESCRIPTION
**Pull request by tadelesh**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

resolve: https://github.com/Azure/cadl-ranch/issues/251

### CODE REVIEW

**Comment by tadelesh** [comment]
resolve: https://github.com/Azure/cadl-ranch/issues/251

File: packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts [hunk 1]
```diff
@@ -91,3 +91,14 @@ Scenarios.Type_Property_ValueTypes_DictionaryString_put = passOnSuccess(dictiona
 const neverMock = createMockApis("never", undefined);
 Scenarios.Type_Property_ValueTypes_Never_get = passOnSuccess(neverMock.get);
 Scenarios.Type_Property_ValueTypes_Never_put = passOnSuccess(neverMock.put);
+
+const recordUnknownMock = createMockApis("record/unknown", {
```
Comment (timotheeguerin) [inline packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts:95]: I think this test should go in the dictionary test, this is basically that.

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts:95]: Oh, I see. There already exists a test case of `Record<unknown>` under dictionary. I will close this PR.

**Comment by tadelesh** [comment]
Currently, there is a lint rule that disallow `Record<unknown>`: https://github.com/Azure/typespec-azure/blob/8c23a8e9a2bd3fe7462560370f6e0ea15332bb72/packages/typespec-azure-core/src/rules/record-types.ts#L5. If we need to support any JSON object using `Record<unknown>`, we should remove such lint.

File: packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts (hunk 1 above)
Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts:95]: yeah it is a bit different but the goal of this test is to test different types for a model property
while the other is to test the different types for the value of a record.

So in this test we want to test one Record but we don't need to test every single record combination on top.
This shouldn't make any difference in the generation no?

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts:95]: Do you mean the we only need one Record<> test for type-property test and left tests for all kinds of types for Record value in dictionary test? If so, I'm OK to close this PR.

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/type/property/value-types/mockapi.ts:95]: yep, I don't think it brings extra value

**Pull request closed without merging**